tools/gyp/pylib/gyp/generator/make.py

```diff
@@ -2227,3 +2234,10 @@ def CalculateMakefilePath(build_file, base_name):
   root_makefile.write(SHARED_FOOTER)
 
   root_makefile.close()
+
+  # Hack to get rid of $(obj)/path/to/foo.o deps that node.gyp adds manually.
```

Comment:
Should we float the gyp changes in a separate commit in case we lose them in a future gyp update?

Reply:
Good idea.  I'll split it out before landing.